ansys/tools/repo_sync/repo_sync.py

```diff
@@ -34,48 +34,21 @@ def synchronize(
 
     # Create a temporary folder
     with tempfile.TemporaryDirectory() as temp_dir:
-        os.chdir(temp_dir)
-
         # Clone the repo.
-        process = subprocess.Popen(
-            ["git", "clone", f"https://{token}@github.com/{organization}/{repository}"],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
-        stdout, stderr = process.communicate()
+        subprocess.check_call(["git", "clone", f"https://{token}@github.com/{organization}/{repository}"], cwd=temp_dir)
 
-        os.chdir(repository)
+        repo_path = os.path.join(temp_dir, repository)
 
         # Set credential
-        process = subprocess.Popen(
-            ["git", "config", "user.name", f"{user_name}"],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
-        stdout, stderr = process.communicate()
+        subprocess.check_call(["git", "config", "user.name", f"{user_name}"], cwd=repo_path)
 
-        process = subprocess.Popen(
-            ["git", "config", "user.email", f"{user_email}"],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
-        stdout, stderr = process.communicate()
+        subprocess.check_call(["git", "config", "user.email", f"{user_email}"], cwd=repo_path)
 
         # Create a new branch.
         try:
-            process = subprocess.Popen(
-                ["git", "checkout", "-b", branch_name],
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
-            )
-            stdout, stderr = process.communicate()
+            subprocess.check_call(["git", "checkout", "-b", branch_name], cwd=repo_path)
         except:
-            process = subprocess.Popen(
-                ["git", "checkout", branch_name],
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
-            )
-            stdout, stderr = process.communicate()
+            subprocess.check_call(["git", "checkout", branch_name], cwd=repo_path)
 
         # Read manifest
         with open(manifest, "r") as f:
```

Comment:
Is the manifest path supposed to be relative to the call location, or the cloned repository? Here, I was unsure if removing the `chdir` may have unintended consequences.

Reply:
It is relative to the call location.